src/plugin/file/shp/shpdescriptor.js

```diff
@@ -116,6 +118,25 @@ plugin.file.shp.SHPDescriptor.prototype.clearData = function() {
 };
 
 
+/**
+ * @inheritDoc
+ */
+plugin.file.shp.SHPDescriptor.prototype.getExporter = function() {
+  return new plugin.file.shp.SHPExporter();
+};
+
+
+/**
+ * @inheritDoc
+ */
+plugin.file.shp.SHPDescriptor.prototype.onDataChange = function(options) {
+  plugin.file.shp.SHPDescriptor.base(this, 'onDataChange', options);
+
+  // ensure that the URL is set correctly in case we are converting from a shp/dbf file to a zip shp file
+  this.setUrl2('');
```

Comment:
Is this the best place to make this correction, or could we do it closer to where the shp/dbf -> zip actually happens?

Reply:
I'm not sure that I know where that place would be. The SHP exporter isn't aware of the underlying file structure backing a layer when it creates the ZIP to export. It's only in the descriptor on this new code path that we would want to ensure that we're not still loading in a DBF file that we no longer need.

Reply:
Ok, as long as we're certain this path will always be saving a ZIP and not blowing away the DBF URL then I don't have any issue putting it here.